File: Vehifleet.Data.Models/Maintenance.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Vehifleet.Data.Models.BaseEntities;

namespace Vehifleet.Data.Models
{
    public class Maintenance : AuditableEntity
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int VehicleId { get; set; }

        public virtual Vehicle Vehicle { get; set; }
        
        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }        

        [Required]
        public string Description { get; set; }

        [Required]
        public int Mileage { get; set; }

        [Required]
        public bool Regular { get; set; }

        [Required]
        public bool Completed { get; set; }

        [Required]
        [Column(TypeName = "decimal(16, 2)")]
        public decimal Cost { get; set; }
    }
}

File: Vehifleet.Client/src/app/classes/maintenance/maintenance.ts
import { Auditable } from '../base/auditable';

export class Maintenance extends Auditable {
  id: number = 0;
  vehicleId: number;
  startDate: Date;
  endDate: Date;
  description: string;
  mileage: number;
  regular: boolean;
  completed: boolean;
  cost: number;

  constructor(id?: number) {
    super();
    this.id = id;
  }
}
